From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Kilmer <eric.d.kilmer@gmail.com>
Date: Mon, 4 Mar 2024 11:20:45 -0500
Subject: [PATCH] 6276: Fix C++ sleighexample

* We need to link to zlib now

* The method for loading compiled sleigh files has changed
---
 .../Features/Decompiler/src/decompile/cpp/sleighexample.cc  | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/sleighexample.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/sleighexample.cc
index 98a25c8ca5..4e627c768f 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/sleighexample.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/sleighexample.cc
@@ -315,12 +315,12 @@ int main(int argc,char **argv)
   ContextInternal context;
 
   // Set up the assembler/pcode-translator
-  string sleighfilename = "specfiles/x86.sla";
+  istringstream sleighfilename("<sleigh>specfiles/x86.sla</sleigh>");
   Sleigh trans(&loader,&context);
 
   // Read sleigh file into DOM
   DocumentStorage docstorage;
-  Element *sleighroot = docstorage.openDocument(sleighfilename)->getRoot();
+  Element *sleighroot = docstorage.parseDocument(sleighfilename)->getRoot();
   docstorage.registerTag(sleighroot);
   trans.initialize(docstorage); // Initialize the translator
 
@@ -356,7 +356,7 @@ int main(int argc,char **argv)
 --# libraries
 --INCLUDES=-I./src/include
 --
---LNK=src/libsla.a
+--LNK=src/libsla.a -lz
 --
 --libsla.a:
 --	$(MAKE) -C src/ $@
-- 
2.44.0

